lib/stack_master/parameter_resolvers/ami_finder.rb

```diff
@@ -0,0 +1,34 @@
+module StackMaster
+  module ParameterResolvers
+    class AmiFinder
+      def initialize(region)
+        @region = region
+        @owner = ['self']
+      end
+
+      def build_filters(value, prefix = nil)
+        value.split(',').map do |name_with_value|
+          name, value = name_with_value.strip.split('=')
+          @owner.push name if name == 'owner_id'
```

Comment:
What I mean is that `@owner` is initialized with `['self']`, and then here you're pushing onto `@owner`, so it now contains two values.  If it overrides the filter, then the end result of this will use both `self` and the passed-in `owner_id` (if I'm not missing something).